src/Conversion/ONNXToTOSA/ConvertONNXToTOSA.cpp

```diff
@@ -12,54 +12,26 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "mlir/Dialect/Tosa/IR/TosaOps.h"
-#include "mlir/Pass/Pass.h"
-#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
-
-#include "src/Dialect/ONNX/DialectBuilder.hpp"
-#include "src/Dialect/ONNX/ONNXOps.hpp"
-#include "src/Dialect/ONNX/ONNXOpsHelper.hpp"
-#include "src/Pass/Passes.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
 
 using namespace mlir;
 
 namespace onnx_mlir {
 
-// This defines a template to construct ops whose legalizations are
-// specialized.
-template <typename OnnxOpT>
-class ConvertOnnxOp : public OpConversionPattern<OnnxOpT> {
-public:
-  using OpConversionPattern<OnnxOpT>::OpConversionPattern;
-  using OpAdaptor = typename OnnxOpT::Adaptor;
-  LogicalResult matchAndRewrite(OnnxOpT op, OpAdaptor adaptor,
-      ConversionPatternRewriter &rewriter) const override;
-};
-
-template <>
-LogicalResult ConvertOnnxOp<ONNXReluOp>::matchAndRewrite(ONNXReluOp op,
-    OpAdaptor adaptor, ConversionPatternRewriter &rewriter) const {
-  Value input = adaptor.X();
-  auto inputTy = input.getType().dyn_cast<TensorType>();
-
-  if (!inputTy)
-    return op.emitError("Only Tensor types supported in TOSA");
-
-  if (!inputTy.getElementType().isa<FloatType>()) {
-    return op.emitError(
-        "Only floating-point datatype legalization currently supported");
-  }
+inline bool isa_tosa_signed_int(Type type) {
```

Comment:
Naming convention is a bit weird. Also, usually no `inline` in `.cpp`, and definitely use `static` instead.